moderation/register.py

```diff
@@ -142,9 +143,9 @@ def pre_save_handler(self, sender, instance, **kwargs):
             moderated_obj = self._get_or_create_moderated_object(instance,
                                                                  unchanged_obj,
                                                                  moderator)
-            if moderated_obj.moderation_status != \
-               MODERATION_STATUS_APPROVED and \
-               not moderator.bypass_moderation_after_approval:
+            if not (moderated_obj.moderation_status ==
+                    MODERATION_STATUS_APPROVED and
+                    moderator.bypass_moderation_after_approval):
```

Comment:
@adityar7: Did you mean to change this logic?

According to [De Morgan's Theorem](https://en.wikipedia.org/wiki/De_Morgan%27s_laws):

> "NOT (A *OR* B)" is the same as "(NOT A) AND (NOT B)"

rearranging that:

> "(NOT A) AND (NOT B)" is the same as "NOT (A *OR* B)"

(emphasis mine)

|   mod_status  | bypass moderation | previous result | new result |     | not (A *or* B)         | -->       | result |
| ------------- | ----------------- | --------------- | ---------- | --- | ------------ | --------- | ------ |
| NOT APPROVED  | False             | True            | True       |     | not (False or False) | not False | True   |
| NOT APPROVED | True              | **False**           | **True**      |     | not (False or True)  | not True  | False  |
| APPROVED     | False             | **False**           | **True**       |     | not (True or False)  | not True  | False  |
| APPROVED     | True              | False           | False      |     | not (True or True)   | not True  | False  |

I think this logic should be:

```python
           if not (moderated_obj.moderation_status ==
                    MODERATION_STATUS_APPROVED or  # <-- OR here
                    moderator.bypass_moderation_after_approval):
```